server/protocols/webadmin/webadmin-mailqueue/src/test/java/org/apache/james/webadmin/routes/MailQueueRoutesTest.java

```diff
@@ -477,55 +512,115 @@ public void forcingDelayedMailsDeliveryShouldRejectRequestWithoutBody() throws E
     @Test
     public void deleteMailsShouldDeleteMailsWhenSenderIsGiven() throws Exception {
         MemoryMailQueue queue = mailQueueFactory.createQueue(FIRST_QUEUE);
-        String sender = "sender@james.org";
-        queue.enQueue(Mails.defaultMail()
-                .sender(sender)
-                .build());
+
+        queue.enQueue(FakeMail.builder()
+            .name(FAKE_MAIL_NAME_1)
+            .sender(SENDER_1_JAMES_ORG)
+            .build());
+
+        queue.enQueue(FakeMail.builder()
+            .name(FAKE_MAIL_NAME_2)
+            .sender(SENDER_2_JAMES_ORG)
+            .build());
+
+        String taskId = with()
+            .param("sender", SENDER_1_JAMES_ORG)
+            .delete(FIRST_QUEUE + "/mails")
+            .jsonPath()
+            .getString("taskId");
 
         given()
-            .param("sender", sender)
+            .basePath(TasksRoutes.BASE)
         .when()
-            .delete(FIRST_QUEUE + "/mails")
+            .get(taskId + "/await")
         .then()
-            .statusCode(HttpStatus.NO_CONTENT_204);
-
-        assertThat(queue.browse()).isEmpty();
```

Comment:
queue.browse seems to be a better way of checking which mail had been deleted.

Maybe we can browse the queue, to get the names of the contained email, and then assertThat we only have FAKE_MAIL_NAME_2 ?